nav2_collision_monitor/src/polygon_base.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/polygon_base.hpp"
+#include "nav2_collision_monitor/kinematics.hpp"
+
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_collision_monitor
+{
+
+PolygonBase::PolygonBase(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  const std::string & polygon_name,
+  const std::string & base_frame_id,
+  const double simulation_time_step)
+: node_(node), polygon_name_(polygon_name), base_frame_id_(base_frame_id),
+  action_type_(DO_NOTHING), max_points_(-1), slowdown_ratio_(0.0), time_before_collision_(-1.0),
+  simulation_time_step_(simulation_time_step), visualize_(false)
+{
+  RCLCPP_INFO(logger_, "[%s]: Creating PolygonBase", polygon_name_.c_str());
```

Comment:
Base class doesn't need the create / destroy logs; the implementations have them with more information

Reply:
In the latest update all logs in constructors/destructors were removed for simplicity